pkg/kubelet/rkt/rkt.go

```diff
@@ -0,0 +1,391 @@
+	Garbage        = "garbage"
+)
+
+const (
+	dockerAuthTemplate = `{"rktKind":"dockerAuth","rktVersion":"v1","registries":[%q],"credentials":{"user":%q,"password":%q}}`
+)
+
+// Runtime implements the ContainerRuntime for rkt. The implementation
+// uses systemd, so in order to run this runtime, systemd must be installed
+// on the machine.
+type Runtime struct {
+	systemd *dbus.Conn
+	absPath string
+	config  *Config
+	// TODO(yifan): Refactor this to be generic keyring.
+	dockerKeyring credentialprovider.DockerKeyring
+}
+
+// New creates the rkt container runtime which implements the container runtime interface.
+// It will test if the rkt binary is in the $PATH, and whether we can get the
+// version of it. If so, creates the rkt container runtime, otherwise returns an error.
+func New(config *Config) (*Runtime, error) {
+	systemdVersion, err := getSystemdVersion()
+	if err != nil {
+		return nil, err
+	}
+	result, err := systemdVersion.Compare(systemdMinimumVersion)
+	if err != nil {
+		return nil, err
+	}
+	if result < 0 {
+		return nil, fmt.Errorf("rkt: systemd version is too old, requires at least %v", systemdMinimumVersion)
+	}
+
+	systemd, err := dbus.New()
+	if err != nil {
+		glog.Errorf("rkt: Cannot connect to dbus: %v", err)
+		return nil, err
+	}
+
+	// Test if rkt binary is in $PATH.
+	absPath, err := exec.LookPath(rktBinName)
+	if err != nil {
+		glog.Errorf("rkt: Cannot find rkt binary: %v", err)
+		return nil, err
+	}
+
+	rkt := &Runtime{
+		systemd:       systemd,
+		absPath:       absPath,
+		config:        config,
+		dockerKeyring: credentialprovider.NewDockerKeyring(),
+	}
+
+	// Test the rkt version.
+	version, err := rkt.Version()
+	if err != nil {
+		return nil, err
+	}
+	result, err = version.Compare(rktMinimumVersion)
+	if err != nil {
+		return nil, err
+	}
+	if result < 0 {
+		return nil, fmt.Errorf("rkt: Version is too old, requires at least %v", rktMinimumVersion)
+	}
+	return rkt, nil
+}
+
+func (r *Runtime) buildCommand(args ...string) *exec.Cmd {
+	cmd := exec.Command(rktBinName)
+	cmd.Args = append(cmd.Args, r.config.buildGlobalOptions()...)
+	cmd.Args = append(cmd.Args, args...)
+	return cmd
+}
+
+// runCommand invokes rkt binary with arguments and returns the result
+// from stdout in a list of strings.
+func (r *Runtime) runCommand(args ...string) ([]string, error) {
+	glog.V(4).Info("rkt: Run command:", args)
+
+	output, err := r.buildCommand(args...).Output()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(string(output)), "\n"), nil
+}
+
+type resource struct {
+	limit   string
+	request string
+}
+
+// setIsolators overrides the isolators of the pod manifest if necessary.
+func setIsolators(app *appctypes.App, c *api.Container) error {
+	var isolator appctypes.Isolator
+	if len(c.Capabilities.Add) > 0 || len(c.Capabilities.Drop) > 0 || len(c.Resources.Limits) > 0 || len(c.Resources.Requests) > 0 {
+		app.Isolators = []appctypes.Isolator{}
```

Comment:
nope, the app is from the imageManifest, so it can have some non-empty isolators.
What I am doing here is to override the isolators if any isolators are specified in the pod spec